### DISCUSSION TITLE
Is it possible to detect if Scene Viewer will work on a specific android device in the most resent versions of model viewer?

### DISCUSSION THREAD

**Discussion by hd-esteban** [google/model-viewer#2426] [General]
As far as I saw in some older implementations of model-viewer it was not possible to identify whether the android device would be able to open Scene viewer or not. It used a fallback_url to redirect to in case of any error and a has was used to force the Full screen view for that cases.

For ARQuickLook it is now possible to detect this if the tag <a> supports the AR value for the attribute and show/hidde the ar implementations if it is not able?

is this possible in the new AR core/Scene viewer versions?

**Comment by elalish** [comment]
No, Quick Look is able to do that because they're hacked directly into Safari. Chrome doesn't allow that kind of exception, partially to protect privacy against fingerprinting.

**Reply by hd-esteban** [reply]
Thank you Elalish, any workaround?

**Reply by elalish** [reply]
we have the workarounds you've noted. If you want to work around Chrome's privacy, I think that's unlikely. You get a somewhat more accurate signal from WebXR, but not perfect either.

**Reply by hd-esteban** [reply]
yes, I know, just one more question, do you have something like a flow chart where we can see the complete flow to show/hide the ar button on each type of device/os?

**Reply by elalish** [reply]
https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/features/ar.ts#L213-L242

**Reply by hd-esteban** [reply]
Thank you, this is what i need to have an idea of the supported devices, just one more question for the Scene viewer candidates I noticed you has a validation to show the AR button in case it is android but not firefox or oculus, with this validation in which browsers will it work besides google chrome?

**Comment by ghost** [comment]
Perhaps you're looking for the is-ar-supported sample?
https://modelviewer.dev/examples/augmentedreality/

**Reply by hd-esteban** [reply]
Hi @timmmeh  thanks for the info, it also helps me a lot, but in my last question I was asking why here: https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/features/ar.ts#L213-L242 they're validating that it is Android and not oculus or firefox instead of validating directly that it is Android and chrome to define if it is a scene viewer candidate and show the AR button.

**Reply by hd-esteban** [reply]
@elalish Is there any other browser that supports intent links?

**Reply by elalish** [reply]
Maybe the Chromium-based ones? I'm not sure, you may have to test.

**Reply by hd-esteban** [reply]
thank you for your feedback :) I'll test it
